### ISSUE TITLE
nan_json.h(89): error C2660: 'v8::JSON::Parse' : function does not take 2 arguments

### ISSUE THREAD

**Issue by YLivay** [nodejs/nan#670]
Hi, 
since around a month ago, we've been having issues building native modules for nwjs.
The problematic dependency is lzma-native which depends on nan@^2.3.2.

The error logs I'm seeing seems to be related to recent breaking changes with json parsing:
```
c:\projects\gamejolt\build\client\prod\source\node_modules\nan\nan_json.h(89): error C2660: 'v8::JSON::Parse' : function does not take 2 arguments [C:\projects\gamejolt\build\client\prod\source\node_modules\lzma-native\build\lzma_native.vcxproj]
```
Is this related in some way to https://github.com/nodejs/nan/issues/658?

**Comment by bnoordhuis** [comment]
In a way.  I suspect it's because nw.js announces (through the NODE_MODULE_VERSION macro) that it's compatible with node.js v6 when in fact it ships a newer version of V8 than ships with the real node.js v6.

I don't know of a way to reliably distinguish between node.js and nw.js so this might be unfixable in nan.  It wouldn't be an issue if nw.js had its own NODE_MODULE_VERSION but it doesn't, as far as I can tell.

**Comment by mkrufky** [comment]
It might be worth the test of forcing usage of older nan version 2.6.1, before `Fix v8::JSON::Parse() deprecation warning.` got applied.  Before that commit, we used a single argument version of `v8::JSON::Parse`, which will generate warnings against some node versions but should continue to build successfully.

If this works for you, it will be good for us to know, but like Ben said, I'm not sure of the best way to fix it moving forward.

**Comment by mkrufky** [comment]
I believe this issue is resolved in _nan_ version `2.8.0` ... It would be really nice if somebody can confirm that and then we can close this issue.....

**Comment by YLivay** [comment]
Hey @mkrufky thanks a ton for looking into it, and sorry for not replying earlier. In the frenzy of trying to find a workaround i forgot to check back on the issue :(

Our build no longer depends on native modules and setting up the old environment is too time consuming atm, but i'll get to it and update back here as soon as I can. You can close the issue if you think it's fixed, i'll recreate it if the issue persists.
